tools/miq_config_sssd_ldap/sssd_conf.rb

```diff
@@ -42,7 +42,7 @@ def create
       LOGGER.debug("Invoked #{self.class}##{__method__}")
 
       begin
-        File.write(dest_path, ERB.new(File.read(src_path), nil, '-').result(binding))
+        File.write(dest_path, ERB.new(File.read(src_path), :trim_mode => '-').result(binding))
```

Comment:
So, ruby 3.0.7 has 2.2.0 which has this interface:

```
  def initialize(str, safe_level=NOT_GIVEN, legacy_trim_mode=NOT_GIVEN, legacy_eoutvar=NOT_GIVEN, trim_mode: nil, eoutvar: '_erbout')
```

I dropped the safe_level and passed a trim_mode so it looks like it's a bridge interface to allow legacy interface with positional arguments.

Note, ruby 3.0.0 has the same erb interface.

Reply:
@Fryguy see above...

https://github.com/ruby/ruby/blob/v3_0_7/lib/erb.rb#L809
https://github.com/ruby/ruby/blob/v3_0_0/lib/erb.rb#L809

Reply:
So, I think https://github.com/ManageIQ/manageiq-appliance_console/pull/267.  It looks like the interface was changed in ruby 2.6 to accept either kwargs or positional arg for trim_level.  Safe level can be excluded safely

Reply:
Yeah I;m so lost - when I researched this the other day, I could have sworn that 2.2.0 (from Ruby 3.0) didn't have the keyword args interface.